### ISSUE TITLE
"Up a Level" in File Manager Not Working

### ISSUE THREAD

**Issue by alchokem** [TeamWin/Team-Win-Recovery-Project#1534]
- [X ] I am running an official build of TWRP, downloaded from https://twrp.me/Devices/
- [X ] I am running the latest version of TWRP
- [ X] I have read the FAQ (https://twrp.me/FAQ/)
- [ X] I have searched for my issue and it does not already exist

**Device codename**: <HTC 10>
**TWRP version**: <3.3.1>

#### WHAT STEPS WILL REPRODUCE THE PROBLEM?
Boot Into TWRP Recovery -> Advanced -> File Manager -> Select any Folder (Folder Contents is Displayed) -> Select "Up A Level"

#### WHAT IS THE EXPECTED RESULT?
Should Navigate back up the Folder Tree

#### WHAT HAPPENS INSTEAD?
Nothing at all.

#### ADDITIONAL INFORMATION
 Happens with older versions as well.

`/tmp/recovery.log`: 
`dmesg`:

**Comment by CaptainThrowback** [comment]
Can you post about this issue in the TWRP thread and confirm others are seeing this? Are you using a custom theme or anything? I don't have a 10 to test on presently, but if this is an issue on the 10, I would think it would be an issue on other devices, and it definitely isn't.

**Comment by alchokem** [comment]
No custom theme. I've never seen this behavior before either on any device I've installed TWRP on and I have used TWRP for years (on several HTC 10's as well).  This problem showed up just after I had the battery replaced on a specific HTC 10 by a local repair shop.  I don't know why having the battery replaced would have any effect but that is exactly when the problem surfaced.  Thanks for your assistance. Any further insight would be greatly appreciated. 

N.B. I would also add that a second similar failure is also occurring..I also cannot select a specific backup to restore that I had created earlier with TWRP.  The common thread tying both problems together - "Up A Level" and selection of the TWRP backup They both appear on the first lines of their respective screens (File Manager and Restore).

**Comment by alchokem** [comment]
One more observance:  I can't select "Reboot after installation is complete" (second Line in Settings -> General).If I scroll the display up one row, I can select "Reboot....". I also cannot select "UTC-11..." under Settings -> Time Zone.   With the phone fully booted to Oreo, there is no discernible "dead spots" on the screen.

**Comment by CaptainThrowback** [comment]
Did you post in the TWRP support thread on XDA like I asked? This doesn't appear to be a TWRP issue, but specific to your device. I also don't see any logs posted in case something in there provides a clue on the issue.

This can remain open until we can confirm this is only happening on your device, which having others with your device to test will confirm. I can confirm that I'm not seeing this behavior on any of my devices, which suggests it's either specific to the 10, or to you. Posting in the thread should help us to confirm which is the case.

**Comment by alchokem** [comment]
Not familiar with XDA and TWRP Support thread. Can you provide me with a URL where I should post this issue? In reviewing this issue at XDA, I found multiple posts with very similar issues being reported. In 2016, two individuals reported TWRP's touchscreen not working after replacing their screens on HTC 10's. In 2019, several TWRP touchscreen not working issues were reported on Xiaomi Mi devices. Seems to me this problem isn't limited to just my phone. There are lots of posts about the touch screen in TWRP not working. Simply search TWRP and Touchscreen to see what I mean

**Comment by CaptainThrowback** [comment]
If you're saying this is a touchscreen issue, I'm not sure what you want me to do. It's not a problem with OEM touchscreens, just with aftermarket ones. That's not something I can fix. And it's certainly not a TWRP upstream issue. Please close - there's nothing I will be doing about this.

**Comment by alchokem** [comment]
There are lots of posts about the touch screen in TWRP not working. Simply search TWRP and Touchscreen to see what I mean. Not just from those who have replaced screens. More important question...Why have so many people reported the problem is limited to TWRP and that their phones work without issue when fully booted.?

**Comment by CaptainThrowback** [comment]
Not working touchscreen in TWRP is not a TWRP upstream issue. Most of those issues are due to kernel which restrict the ability of the touchscreen to work in recovery. Your issue is completely unrelated to that, since you're focusing on a specific item in File Manager not working. If touch is working for you at all in TWRP, then this isn't your problem. Look on the TWRP device page for the 10 to find the link to the support thread and post your issue there and see if anyone else is having it. Since this isn't an upstream issue, please close this.

**Comment by alchokem** [comment]
Contrary to your statement, my problem isn't limited to File Manager as I explained to you earlier with multiple examples.  It happens in  multiple places... in Settings ..In File Manager...in Restore.  As I said earlier they are tied to a similar physical action of selecting the first lines of each of these screens.  I'm an IT guy with 40 years of experience...appears to be a software issue to me. We both know it is quite possible that a user could interact with multiple items in a list (which all these screens present) and not be able to select the first one. In this case which is my case, it is clearly a software bug. I'm not sure what good posting this issue in the HTC thread again would do since other individuals have reported this problem already on this phone...also as is clearly the case this problem is not limited to the HTC 10.

**Comment by bigbiff** [comment]
TWRP uses an event manager to read events from the touch panel device initialized by the kernel. Sometimes the kernel initializes extra devices that need to be blacklisted from event devices by BoardConfig variables. Sometimes device drivers change, and need a new kernel update. The event source code hasn't changed in years, due to most problems being device side. Sometimes the OEM blocks the touch driver from initializing properly in recovery.

TLDR; the device maintainer probably needs to get your specific conditions to determine the issue.

**Comment by CaptainThrowback** [comment]
@alchokem 

Have you tried one of the unofficial TWRP versions listed [here](https://github.com/CaptainThrowback/android_device_htc_pme/releases)?

Please do that any tell me if you still have this problem. And please close this issue and open it [in the device repo](https://github.com/TeamWin/android_device_htc_pme/issues).

P.S. You still haven't posted any logs. In this case, a video of the issue might help too, because I don't really understand the problem. Have you tried installing a custom theme and see if that makes any difference?

**Comment by alchokem** [comment]
Opened the issue in the device repo as you requested and closed this one.   Don't know how to be any clearer...select the first item in the Restore Screen, the first item in any of the five Settings screens, or "Up A Level" (first item in the File Manager screen) and nothing, I mean nothing happens. Select any item in the list of any of the screens except the first one and everything works as it should.  As a reminder, this all began occurring after I did a factory reset and had the battery replaced by a licensed dealer.  I would also point out that I am running Magisk as well. Why any of these things would cause this behavior is beyond me.  I've run TWRP for years on many devices and have never seen this behavior before.
